code/socialDistribution/templates/tagtemplates/post.html

```diff
@@ -100,6 +100,11 @@ <h4 class="card-title mx-auto">{{ post.title }}</h4>
         </div>
     </div>
 
+    <form method="POST" action="{% url 'socialDistribution:share-post' post.id %}">
+        {% csrf_token %}
+        {% modal id="share_post_modal"|addstr:post.id type="share" label="Share Post" title="Share Post" btn="Share" postid=post.id%}
+    </form>
+
     {% if post_type == 'local' %}
         {% load modal %}
```

Comment:
I think we need to move the form on line 103 after the endif keyword as we have to load the model?